src/Symfony/Component/Yaml/Dumper.php

```diff
@@ -115,6 +115,19 @@ public function dump($input, $inline = 0, $indent = 0, $flags = 0)
                 if ($value instanceof TaggedValue) {
                     $output .= sprintf('%s%s !%s', $prefix, $dumpAsMap ? Inline::dump($key, $flags).':' : '-', $value->getTag());
 
+                    if ($inline >= 1 && Yaml::DUMP_MULTI_LINE_LITERAL_BLOCK & $flags && \is_string($value->getValue()) && false !== strpos($value->getValue(), "\n") && false === strpos($value->getValue(), "\r\n")) {
+                        // If the first line starts with a space character, the spec requires a blockIndicationIndicator
+                        // http://www.yaml.org/spec/1.2/spec.html#id2793979
+                        $blockIndentationIndicator = (' ' === substr($value->getValue(), 0, 1)) ? (string) $this->indentation : '';
+                        $output .= sprintf(" |%s\n", $blockIndentationIndicator);
+
+                        foreach (preg_split('/\n|\r\n/', $value->getValue()) as $row) {
```

Comment:
should be `\r\n|\n`

Reply:
Why don't we just split at `\n`? According to the `if` on line 118 `\r\n` will not be part of the tag value, will it?

Reply:
@nicolas-grekas @xabbuh 

Thank you for your feedback, to be honest this peace of code is only a copy/paste of lines 102:113 of the original `Dumper` with the only difference of getting the value of the `TaggedValue`.

I've tested locally both of your solutions and the tests are green, and yes @xabbuh you're right according to the `if` on line 118 only the values with `\n` and no `\r\n` will go through.

Based on the above I've updated the code to:

- Change `preg_split` pattern on lines 108 and 124 to `/\n/` (normal values and tagged values)
- Create a new test to assert that a tagged value containing `\r\n` will be dumped inline

Reply:
@xabbuh @nicolas-grekas I think this conversation is now outdated because of the latest changes.